applications/asset_tracker_v2/src/cloud/nrf_cloud_integration.c

```diff
@@ -36,6 +40,51 @@ static struct k_work_delayable user_associating_work;
 
 static cloud_wrap_evt_handler_t wrapper_evt_handler;
 
+#if defined(CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE)
+/* Full modem FOTA requires external flash to hold the full modem image.
+ * Below is the external flash device present on the nRF9160 DK version
+ * 1.0.1 and higher.
+ */
+static struct dfu_target_fmfu_fdev ext_flash_dev = {
+	.size = 0,
+	.offset = 0,
+	.dev = DEVICE_DT_GET_ONE(jedec_spi_nor)
+};
+
+static int fmfu_and_modem_init(const struct device *dev)
+{
+	enum nrf_cloud_fota_type fota_type = NRF_CLOUD_FOTA_TYPE__INVALID;
+	int ret;
+
+	ARG_UNUSED(dev);
+
+	/* Check if a full modem FOTA job is pending */
+	ret = nrf_cloud_fota_pending_job_type_get(&fota_type);
+
+	/* Process pending full modem FOTA job before initializing the modem library */
+	if ((ret == 0) && (fota_type == NRF_CLOUD_FOTA_MODEM_FULL)) {
+		/* The flash device must be set before a full modem FOTA update can be applied */
+		ret = nrf_cloud_fota_fmfu_dev_set(&ext_flash_dev);
+		if (ret < 0) {
+			LOG_ERR("Failed to set flash device for full modem FOTA, error: %d",
+				ret);
+		} else {
+			ret = nrf_cloud_fota_pending_job_validate(NULL);
+			if ((ret < 0) && (ret != -ENODEV)) {
+				LOG_ERR("Error validating full modem FOTA job: %d", ret);
```

Comment:
Don't we want to return err if this fails?

Reply:
the FOTA status is saved and will be reported to nrf cloud as a failed FOTA update.
it should not impact device startup.